src/DictVectors/skinnydvec.jl

```diff
@@ -0,0 +1,34 @@
+"""
+    SkinnyDVec(keys, vals; style) <: AbstractDVec
+Implementation of an [`AbstractDVec`](@ref) using lists implemented as `Vector`s for
+internal storage. Doesn't comply with the full [`AbstractDVec`](@ref) interface.
+
+Supports fast iteration. Adding elements with `setindex!` just appends them to the list.
+"""
+struct SkinnyDVec{K,V,S<:StochasticStyle{V}} <: AbstractDVec{K,V}
```

Comment:
Should this be merged with `FrozenDVec`?
I'm not sure, since they are doing very different things, but they are implemented similarly.

Reply:
Yes, probably.

I notice that `FrozenDVec` is a vector of pairs, which is probably better as we need access to pairs when iterating. The usage of the two types is complementary, but it looks like everything could be implemented by a single type.